html/semantics/embedded-content/the-iframe-element/iframe-loading-lazy-history-pushState.html

```diff
@@ -0,0 +1,22 @@
+<!DOCTYPE html>
+<title>History state change for iframe loading='lazy' before it is loaded: history.pushState</title>
+<iframe data-src="support/blank.htm?foo" src="support/blank.htm?src" loading="lazy"></iframe>
+<script>
+const iframe = document.querySelector('iframe');
+// Should have no effect on lazy-loading
+iframe.contentWindow.history.pushState(null, "", iframe.dataset.src);
+</script>
+<!-- Loading testharness.js here is intentional. -->
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script>
+setup({single_test: true});
+onload = () => {
+  step_timeout(() => {
+    assert_equals(iframe.contentWindow.location.href, new URL("support/blank.htm?src", location.href).href);
```

Comment:
Should we assert above this, that `location.href` equals `support/blank.htm?foo` (at least I think it will at this time?)

Reply:
Right after calling `pushState` yes. When this script runs, it's racy whether the lazy load has happened.

Reply:
The `pushState` and `replaceState` tests were incorrect since the iframe's doc's URL is `about:blank`, and so the URL components (except fragment) need to match `about:blank`. Fixed now.

Reply:
> When this script runs, it's racy whether the lazy load has happened.

What do you mean by "happened"? Do you mean https://html.spec.whatwg.org/multipage/iframe-embed-object.html#the-iframe-element:current-navigation-was-lazy-loaded-4 just being set to true? Or do you mean that _plus_ the resumption steps executed, and the `support/blank.htm?src` navigation has completed?

In any case, I'm not sure I follow. By this point in the script, we've done nothing asynchronous, right? I think what has happened up to this point is:

1. https://html.spec.whatwg.org/multipage/iframe-embed-object.html#the-iframe-element:current-navigation-was-lazy-loaded-4 gets set to true, for the lazyload navigation to `support/blank.html?src`. But the iframe's document is still `about:blank`, since the lazy load resumption steps were not run immediately
2. Synchronously after, `pushState()` comes along and [changes](https://html.spec.whatwg.org/multipage/browsing-the-web.html#url-and-history-update-steps:~:text=Set%20document%27s%20URL%20to%20newURL) the `about:blank` Document's URL to `about:blank#foo`, which is allowed per https://html.spec.whatwg.org/C#can-have-its-url-rewritten

So is there any room in there for the lazy loaded navigation to have already completed by then?

Reply:
The navigation might have completed. The event loop is allowed to spin between each script, and the `pushState()` happens in the first script element.